### PR TITLE
Grasp candidate config param

### PR DESCRIPTION
**Pull request by mlautman**
This PR adds more control over which types of grasps are generated
See last commit for new changes.

### CODE REVIEW

File: include/moveit_grasps/grasp_generator.h [hunk 1]
```diff
@@ -101,32 +101,65 @@ struct GraspCandidateConfig
...
+  GraspCandidateConfig(const GraspCandidateConfig& config)
+    : enable_corner_grasps_(config.enable_corner_grasps_)
+    , enable_face_grasps_(config.enable_face_grasps_)
+    , enable_variable_angle_grasps_(config.enable_variable_angle_grasps_)
+    , enable_edge_grasps(config.enable_edge_grasps)
+    , generate_x_axis_grasps_(config.generate_x_axis_grasps_)
+    , generate_y_axis_grasps_(config.generate_y_axis_grasps_)
+    , generate_z_axis_grasps_(config.generate_z_axis_grasps_)
+  {
+  }
```
Comment (davetcoleman) [inline include/moveit_grasps/grasp_generator.h]: remove this redundant copy constructor, "the implicitly generated one is just a convenience that does “the right thing” for POD types:"

https://stackoverflow.com/questions/12979529/default-constructors-and-pod

File: include/moveit_grasps/grasp_generator.h [hunk 2]
```diff
@@ -101,32 +101,65 @@ struct GraspCandidateConfig
...
+    generate_x_axis_grasps_ = true;
+    generate_y_axis_grasps_ = true;
+    generate_z_axis_grasps_ = true;
+  }
+  void enableAll()
+  {
+    enableAllTypes();
+    enableAllAxes();
+  }
+  void disableAllTypes()
```
Comment (davetcoleman) [inline include/moveit_grasps/grasp_generator.h]: disableAllGraspTypes

also this "struct" sure is getting complicated...

File: src/grasp_planner.cpp [hunk 3]
```diff
@@ -87,11 +87,11 @@ bool GraspPlanner::planAllApproachLiftRetreat(std::vector<GraspCandidatePtr>& gr
...
     else
     {
-      //++grasp_it; // move to next grasp
+      ++grasp_it;  // move to next grasp
 
       // Once we have one valid path, just quit so we can use that one
-      ROS_INFO_STREAM_NAMED("grasp_planner", "Valid grasp plan generated");
-      break;
+      // ROS_INFO_STREAM_NAMED("grasp_planner", "Valid grasp plan generated");
+      // break;
```
Comment (davetcoleman) [inline src/grasp_planner.cpp]: remove commented code

File: src/grasp_planner.cpp [hunk 4]
```diff
@@ -319,7 +319,7 @@ bool GraspPlanner::computeCartesianWaypointPath(const moveit::core::JointModelGr
                                                                << " number of segments in trajectory: "
                                                                << segmented_cartesian_traj.size());
 
-    double min_allowed_valid_percentage = 0.9;
+    double min_allowed_valid_percentage = 0.99;
```
Comment (davetcoleman) [inline src/grasp_planner.cpp]: as discussed

File: include/moveit_grasps/grasp_generator.h (hunk 1 above)
Comment (mlautman) [reply include/moveit_grasps/grasp_generator.h]: cool!

**Pull request merged**